src/platform/STM32/startup/system_stm32g4xx.c

```diff
@@ -371,10 +370,18 @@ void SystemClock_Config(void)
 
   // Initializes the CPU, AHB and APB busses clocks
 
-  RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI|RCC_OSCILLATORTYPE_HSI48
-                              |RCC_OSCILLATORTYPE_LSI|RCC_OSCILLATORTYPE_HSE;
-
-  RCC_OscInitStruct.HSEState = RCC_HSE_ON;
+  uint32_t hse_value = persistentObjectRead(PERSISTENT_OBJECT_HSE_VALUE);
+  if(hse_value == 0)
+  {
+    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI|RCC_OSCILLATORTYPE_HSI48|RCC_OSCILLATORTYPE_LSI;
+    RCC_OscInitStruct.HSEState = RCC_HSE_OFF;
+  }
+  else
+  {
+    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI|RCC_OSCILLATORTYPE_HSI48|RCC_OSCILLATORTYPE_LSI|RCC_OSCILLATORTYPE_HSE;
+    RCC_OscInitStruct.HSEState = RCC_HSE_ON;
+  }
+ 
```

Comment:
IMO this is slightly cleaner (and avoid code duplication):
```suggestion
  const bool useHse = persistentObjectRead(PERSISTENT_OBJECT_HSE_VALUE) != 0;
    RCC_OscInitStruct.OscillatorType = RCC_OSCILLATORTYPE_HSI | RCC_OSCILLATORTYPE_HSI48 
        | RCC_OSCILLATORTYPE_LSI 
        | (useHse ? RCC_OSCILLATORTYPE_HSE : 0);
    RCC_OscInitStruct.HSEState = useHse ? RCC_HSE_ON : RCC_HSE_OFF;
 
```

Reply:
I'm a bit torn on this one but why not...

Reply:
Thanks.

Problem with is is when someone tries to refactor it further - nontrivial part of code is duplicated and it can get sout-of-sync.

Breaking the lines is not necessary, but personally I prefer 80 column limit (unless it is making code harder to read)